src/backend/main.ts

```diff
@@ -1141,6 +1141,56 @@ ipcMain.on('logError', (e, err) => logError(err, LogPrefix.Frontend))
 ipcMain.on('logInfo', (e, info) => logInfo(info, LogPrefix.Frontend))
 
 let powerDisplayId: number | null
+const gamePlaySessionStartTimes: Record<string, bigint> = {}
+
+function startNewPlaySession(appName: string) {
+  // Uses hrtime for monotonic timer not subject to clock drift or sync errors
+  const startPlayingTimeMonotonic = hrtime.bigint()
+  gamePlaySessionStartTimes[appName] = startPlayingTimeMonotonic
+}
+
+async function syncPlaySession(appName: string, runner: Runner) {
+  const stopPlayingTimeMonotonic = hrtime.bigint()
+  const sessionPlaytimeInMs =
+    (stopPlayingTimeMonotonic - gamePlaySessionStartTimes[appName]) /
+    BigInt(1000000)
+
+  // reset the time counter
+  startNewPlaySession(appName)
+
+  // update local json with time played
+  const sessionPlaytimeInMinutes =
+    sessionPlaytimeInMs / BigInt(1000) / BigInt(60)
+
+  const totalPlaytime =
+    sessionPlaytimeInMinutes + BigInt(tsStore.get(`${appName}.totalPlayed`, 0))
+  tsStore.set(`${appName}.totalPlayed`, Number(totalPlaytime))
+
+  const game = gameManagerMap[runner].getGameInfo(appName)
+  const { gameIsEpicForwarderOnHP, hyperPlayListing } =
+    await gameIsEpicForwarderOnHyperPlay(game)
+
+  if (gameIsEpicForwarderOnHP && hyperPlayListing) {
+    postPlaySessionTime(
+      hyperPlayListing.project_id,
+      parseInt((sessionPlaytimeInMs / BigInt(1000)).toString())
```

Comment:
Couldn't you do something like `hyperPlayListing?.project_id || appName` and combine these lines.

Reply:
yup. this simplifies the code a bit